proto/buf/alpha/registry/v1alpha1/labels.proto

```diff
@@ -0,0 +1,68 @@
+// Copyright 2020-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package buf.alpha.registry.v1alpha1;
+
+import "google/protobuf/timestamp.proto";
+
+enum LabelNamespace {
+  LABEL_NAMESPACE_UNSPECIFIED = 0;
+  // This is for globs, '*', in requests. When a label is created
+  // it will be assigned to a specific namespace.
+  LABEL_NAMESPACE_ALL = 1;
+  LABEL_NAMESPACE_TAG = 2;
+  LABEL_NAMESPACE_BRANCH = 3;
+  LABEL_NAMESPACE_GITCOMMIT = 4;
+}
+
+message Label {
+  LabelNamespace namespace = 1;
+  string name = 2;
+  string commit_id = 3;
```

Comment:
Perhaps split the label from its value. commit_id is a label value and needs to be supported in Create, Move, and Get calls. E.g. in Move there's "from" and "to" but they're strings and not describing "from_commit_id" and "to_commit_id". If this value becomes a oneof definition, we'll have a ton of cruft.

```proto
message LabelValue {
  string commit_id = 1;
}
```

Reply:
I'm currently using the `Reference` structure -- do you think we should still define a `LabelValue` structure? It's a bit confusing, because I expect that down the road, these should converge, but for now `Reference` might be too large. I think I've convinced myself to define a separate `LabelValue` message, lol.